### ISSUE TITLE
Make `--settings-file` compatible with flattened JSON like VS Code's `settings.json`

### ISSUE THREAD

**Issue by hdorgeval** [valentjn/languagetool-languageserver#91]
Hi @valentjn, I just tried the new option `--input-documents` and it works great. Thank you so much for providing this feature!

Nonetheless, It is not clear how to use this new option with the `--settings-file` option.

My use case is that I would like to re-use the workspace settings files that have been generated in VSCode:

![Screen Shot 2021-07-25 at 17 08 18](https://user-images.githubusercontent.com/9698924/126904022-d3da6f9b-c3e1-484e-9515-9743e9eddf46.png)

It would be great if you could give some help.

Thanks.

**Comment by valentjn** [comment]
[There is an example in the docs under `--settings-file`.](https://github.com/valentjn/ltex-ls#command-line-arguments)

You have to remove the `ltex.` prefixes to make it work (this stems from the configuration format the LSP mandates). Now that I think about it, you probably also have to unfold everything (that part is missing in the docs). So `{"ltex.latex.commands": ...}` becomes `{"latex": {"commands": ...}}`.

I see why this could confuse people, and it's a good idea to support both possibilities. Keep in mind, though, that VS Code is not the only editor LT<sub>E</sub>X LS supports; other editors might have different setting formats.

**Comment by hdorgeval** [comment]
Hi @valentjn, thanks for your answer. Now I understand more.

For those who will land on this question/issue, here is the excerpt of the settings file that should be given  to the `--settings-file` option (in my use case the input document is in french):

```json
{
  "language": "fr",
  "dictionary": {
    "fr": [
      "Tehemten",
      "Akwan",
      "Rustem",
      "Touran",
      "Raksch",
      "farsangs",
      "Divs",
      // other words omitted for brevity
    ]
  },
  "hiddenFalsePositives": {
    "fr": [
      {
        "rule": "D_N",
        "sentence": "^\\QLorsqu'il fut revenu auprès de la source, méditant dans son âme guerrière de nouvelles luttes, le Div Akwan l'accosta de nouveau et lui dit :\\E$"
      },
      {
        "rule": "D_N",
        "sentence": "^\\QTehemten répondit aux paroles du Div par un mugissement de lion , détacha du crochet son lacet roulé, le lança et prit le Div au milieu du corps.\\E$"
      },
      // other rules omitted for brevity
    ]
  }
}
```

Basically, it is just a copy/paste of the files generated by the [VSCode extension vscode-ltex](https://github.com/valentjn/vscode-ltex)  under the `.vscode` folder.

I will provide soon in this discussion a node-js script that automatically create that settings files from the ones (`ltex.dictionary.fr.txt`, `ltex.hiddenFalsePositives.fr.txt`, `settings.json`) found in the `.vscode`  folder

**Comment by hdorgeval** [comment]
Here is the script that builds the needed settings file from what has been generated by the VSCode extension (this sample code handles only french files):

```ts
  const ltexFrDictionaryFile = path.join(process.cwd(), '.vscode', 'ltex.dictionary.fr.txt');
  const frDictionary = readAllLinesInFile(ltexFrDictionaryFile).filter(
    (line) => line && line.length > 0,
  );

  const frFalsePositiveFile = path.join(
    process.cwd(),
    '.vscode',
    'ltex.hiddenFalsePositives.fr.txt',
  );
  const frFalsePositives = readAllLinesInFile(frFalsePositiveFile)
    .filter((line) => line && line.length > 0)
    .map((rawRule) => JSON.parse(rawRule));

  const frLtexLsSettings = {
    language: 'fr',
    dictionary: { fr: frDictionary },
    hiddenFalsePositives: { fr: frFalsePositives },
  };

  writeFileSync(
    path.join(process.cwd(), 'ltex-settings.fr.json'),
    JSON.stringify(frLtexLsSettings, null, 2),
  );
```

`readAllLinesInFile`:
```ts
export const readAllLinesInFile = (filePath: PathLike): string[] => {
  const lines = readFileSync(filePath, 'utf8').split(/\n|\r/);
  return lines;
};
```

**Comment by valentjn** [comment]
LT<sub>E</sub>X LS can now parse flattened JSON objects as well, with or without `ltex.` prefix. Look in the readme/CLI help for examples.

Note that the settings JSON of the screenshot in the description of this issue won't work though, because that's not valid JSON (has trailing commas). Only valid JSON is supported (no trailing commas, no comments).

**Comment by valentjn** [comment]
Feature released in 13.0.0.
